diff --git a/common_arm.h b/common_arm.h
index 682315de..75e4d776 100644
--- a/common_arm.h
+++ b/common_arm.h
@@ -123,7 +123,7 @@ REALNAME:
 #endif
 #define HUGE_PAGESIZE   ( 4 << 20)
 
-#define BUFFER_SIZE     (32 << 20)
+#define BUFFER_SIZE     (8 << 20)
 
 
 #define BASE_ADDRESS (START_ADDRESS - BUFFER_SIZE * MAX_CPU_NUMBER)
--- a/param.h
+++ b/param.h
@@ -3115,8 +3115,8 @@
 #define CGEMM_DEFAULT_Q 120
 #define ZGEMM_DEFAULT_Q 120
 
-#define SGEMM_DEFAULT_R 12288
-#define DGEMM_DEFAULT_R 8192
+#define SGEMM_DEFAULT_R 4096
+#define DGEMM_DEFAULT_R 4096
 #define CGEMM_DEFAULT_R 4096
 #define ZGEMM_DEFAULT_R 4096
 
@@ -3642,8 +3642,8 @@
 #define CGEMM_DEFAULT_Q 120
 #define ZGEMM_DEFAULT_Q 120
 
-#define SGEMM_DEFAULT_R 12288
-#define DGEMM_DEFAULT_R 8192
+#define SGEMM_DEFAULT_R 4096
+#define DGEMM_DEFAULT_R 4096
 #define CGEMM_DEFAULT_R 4096
 #define ZGEMM_DEFAULT_R 4096
 

